pilot/pkg/config/clusterregistry/multicluster_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterregistry
+
+import (
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	"k8s.io/client-go/kubernetes/fake"
+
+	"istio.io/istio/pkg/kube/secretcontroller"
+
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/istio/pilot/pkg/serviceregistry/aggregate"
+)
+
+const (
+	testSecretName      = "testSecretName"
+	testSecretNameSpace = "istio-system"
+	WatchedNamespace    = "istio-system"
+	DomainSuffix        = "fake_domain"
+	ResyncPeriod        = 1 * time.Second
+)
+
+var mockserviceController = &aggregate.Controller{}
+var clearCacheCalled = false
+
+func mockClearCache() {
```

Comment:
As a note, there is a PR optimizing EDS - in progress in 1.0 to be merged to master. This will remove the use of 'clear cache' for endpoints.

Reply:
@costinm - good to know that will require some changes when a cluster is removed or added regardless of this change set.